pydantic/fields.py

```diff
@@ -341,7 +341,14 @@ def prepare(self) -> None:
         Note: this method is **not** idempotent (because _type_analysis is not idempotent),
         e.g. calling it it multiple times may modify the field and configure it incorrectly.
         """
+
+        # to prevent side effects by calling the default_factory for nothing, we only call it
+        # when we want to validate the default value i.e. when `validate_all` is set to True
+        if self.default_factory is not None and self.type_ is not None and not self.model_config.validate_all:
```

Comment:
I added a check of `self.type_` to avoid regressions but as it's a beta feature I don't know if we could actually just force people to set a type when using it. I'm not sure it makes sense to use `default_factory` without one

Reply:
I think requiring a type with `default_factory` would be a better solution to this.

Reply:
Alright glad we agree on this ! Will remove the extra check then

Reply:
@samuelcolvin Change done in fe533a0